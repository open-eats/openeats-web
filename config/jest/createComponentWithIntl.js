import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';

// See: https://github.com/yahoo/react-intl/wiki/Testing-with-React-Intl#jest

const createComponentWithIntl = (children, props = { locale: 'en' }) => {
  return renderer.create(
    <IntlProvider {...props}>
      {children}
    </IntlProvider>
  );
};

export default createComponentWithIntl;