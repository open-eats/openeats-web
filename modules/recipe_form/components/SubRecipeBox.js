import React from 'react'
import {
  injectIntl,
  defineMessages,
} from 'react-intl';
import ReactTextareaAutocomplete from "@webscopeio/react-textarea-autocomplete";

import SubRecipes from '../../recipe/components/SubRecipes'
import TabbedView from './TabbedView'
import formatQuantity from '../../recipe/utilts/formatQuantity'
import parseIngredient from '../utilts/parseIngredient'

require('../css/smart-text-box.scss');

class SubRecipeBox extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      data: this.props.data || [],
      text: this.unarrayify(this.props.data || []),
    };
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.id !== this.props.id) {
      this.setState({
        data: nextProps.data || [],
        text: this.unarrayify(nextProps.data || []),
      });
    }
  }

  unarrayify = value => {
    let tr = '';
    if (value) {
      // eslint-disable-next-line
      value.map(i => {
        tr += i.numerator ? formatQuantity(1, 1, i.numerator, i.denominator) + " " : '';
        tr += i.measurement ? i.measurement + " " : '';
        tr += i.title + '\n'
      });
    }
    return tr.substring(0, tr.length - 1);
  };

  arrayify = value => {
    let ings = [];
    let subRecipes = value.split('\n').filter(t => t.trim().length > 1);
    for (let index in subRecipes) {
      let line = subRecipes[index];
      if (line.length > 0) {
        ings.push(parseIngredient(line));
      }
    }
    return ings
  };

  handleChange = event => {
    const list = this.arrayify(event.target.value);

    this.setState({
      data: [ ...list ],
      text: event.target.value
    });

    if(this.props.change) {
      this.props.change(this.props.name, [ ...list ]);
    }
  };

  render() {
    const { formatMessage } = this.props.intl;
    const messages = defineMessages({
      info_title: {
        id: 'recipe.create.rf.info_title',
        description: 'info_title',
        defaultMessage: 'Ingredient Help',
      },
      info_desc: {
        id: 'recipe.create.rf.info_desc',
        description: 'info_desc',
        defaultMessage: 'Each Recipe Link should be only its own line. Click the Preview to see what the Recipe Links will look like.',
      },
    });

    const help = {
      infoTitle: formatMessage(messages.info_title),
      infoDesc: formatMessage(messages.info_desc)
    };

    const Item = ({ entity: { name } }) => <div>{`${name}`}</div>;
    const Loading = ({ data }) => <div className="loading">Loading...</div>;

    return (
      <TabbedView { ...{...this.props, ...help} }>
        <div className="form-group">
          <ReactTextareaAutocomplete
            className="form-control"
            loadingComponent={ Loading }
            value={ this.state.text }
            onChange={ this.handleChange }
            rows="4"
            trigger={{
              ":": {
                dataProvider: token => {
                  return this.props.fetchRecipeList(token);
                },
                component: Item,
                output: (item, trigger) => item.char
              }
            }}
          />
        </div>
        <div className="recipe-details">
          <div className="recipe-schema">
            <SubRecipes data={ this.state.data }/>
          </div>
        </div>
      </TabbedView>
    );
  }
}

export default injectIntl(SubRecipeBox)
