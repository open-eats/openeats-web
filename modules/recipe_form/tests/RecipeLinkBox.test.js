import React from 'react';
import SubRecipeBox from '../components/SubRecipeBox';
import createComponentWithIntl from '../../../config/jest/createComponentWithIntl';

test('SubRecipeBox component test', () => {
  const component = createComponentWithIntl(
    <SubRecipeBox/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
