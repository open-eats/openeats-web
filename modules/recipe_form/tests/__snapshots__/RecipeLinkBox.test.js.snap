// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubRecipeBox component test 1`] = `
<div
  className="live-editor"
>
  <ul
    className="nav nav-tabs"
  >
    <li
      className="left active"
    >
      <span>
        Â 
        <span
          className="glyphicon glyphicon-info-sign"
          onClick={null}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        />
      </span>
    </li>
    <li
      className="active"
    >
      <span
        onClick={[Function]}
      >
        Editor
      </span>
    </li>
    <li
      className=""
    >
      <span
        onClick={[Function]}
      >
        Preview
      </span>
    </li>
  </ul>
  <div
    className="content"
  >
    <div
      className="form-group"
    >
      <div
        className="rta  "
        style={undefined}
      >
        <textarea
          className="rta__textarea form-control"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onSelect={[Function]}
          rows="4"
          style={undefined}
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="help-inline"
  />
</div>
`;
